src/Core/python_data_client.py

```diff
@@ -0,0 +1,96 @@
+from typing import Optional, Tuple
+import pandas as pd
+
+class PythonDataClient:
+    def __init__(self):
+        """Client for fetching orderbook/trading data from redis source.
+        
+        TODO: add parameters to configure redis address
+        """
+        pass
+
+    def fetch_trades_by_pair(self, pair: str, n: Optional[int] = 1) -> pd.DataFrame:
```

Comment:
Doesn't need to be optional if you define a default

Also please use modern (3.10+) syntax (`int | None`)

```suggestion
    def fetch_trades_by_pair(self, pair: str, n: int = 1) -> pd.DataFrame:
```